typify-impl/src/lib.rs

```diff
@@ -846,7 +847,8 @@ impl TypeSpace {
                 type_id.clone()
             } else {
                 let type_id = self.assign();
-                self.name_to_id.insert(name.clone(), type_id.clone());
+                ty.ensure_unique_name(self);
+                self.name_to_id.insert(name, type_id.clone());
```

Comment:
do I understand correctly that we're using the old name that might have been changed?